extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -36,57 +36,79 @@ public DevConsoleTemplateInfoBuildItem collectAdditionalEndpoints(
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
 
+                final Stack<StaticResourceInfo.StaticFile> stack = new Stack<>();
+
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
-                        } else {
-                            String parentString = parent.toString();
-                            if (!parentString.startsWith("/")) {
-                                parentString = "/" + parentString;
-                            }
-                            staticResources.add(parentString + "/");
-                        }
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+
+                    Path relativeFile = resource.relativize(dir);
+                    String relativeFileString = relativeFile.toString();
+                    relativeFileString = "/" + relativeFileString;
+                    Path dirName = dir.getFileName();
+                    StaticResourceInfo.StaticFile getStaticFile = staticResourceInfo.resourceMap.get(relativeFile);
+
+                    StaticResourceInfo.StaticFile staticFile = new StaticResourceInfo.StaticFile(dirName.toString(), true,
+                            null);
+                    if (getStaticFile == null) {
+                        staticResourceInfo.resourceMap.put(relativeFile, staticFile);
+                    }
+
+                    // adding to parent
+                    if (!stack.isEmpty()) {
+                        stack.peek().children.add(staticFile);
                     }
-                    if (!file.startsWith("/")) {
-                        file = "/" + file;
+                    stack.push(staticFile);
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    String fileName = file.getFileName().toString();
+                    StaticResourceInfo.StaticFile newStaticFile = new StaticResourceInfo.StaticFile(fileName, false,
+                            null);
+
+                    List<StaticResourceInfo.StaticFile> childrenList = stack.peek().children;
+
+                    if (childrenList == null) {
+                        childrenList = new ArrayList<>();
```

Comment:
Here you're modifying the local variable `childrenList` but not the `StaticFile.children` so it has no effect. If you get rid of the `children` constructor parameter and always have a children list as I mentioned in the other comment you can drop this lazy init.